lib/Component.spec.lua

```diff
@@ -262,6 +267,19 @@ return function()
 		expect(lastProps.bar).to.equal(false)
 
 		Reconciler.unmount(handle)
+
+		lastProps = nil;
+		handle = Reconciler.mount(createElement(TestComponent, {}))
+		Reconciler.reconcile(handle, createElement(TestComponent, {
+			baz = "!",
+		}))
+
+		expect(lastProps).to.be.a("table")
+		expect(lastProps.foo).to.equal("hello")
+		expect(lastProps.bar).to.equal("world")
+		expect(lastProps.baz).to.equal("!")
+
+		Reconciler.unmount(handle)
```

Comment:
Does this test guarantee that `shouldUpdate` got the new props?  `lastProps` gets set by both `shouldUpdate` and `render`, so `render` will be the last one to have modified this variable.

It may be a good idea to break this into a new test.

Reply:
oh you're right, i didn't think about that. i'll split it into a separate case

Reply:
sure (: